v2/pkg/engine/datasource/graphql_datasource/graphql_tws_handler.go

```diff
@@ -259,6 +273,32 @@ func (h *gqlTWSConnectionHandler) handleMessageTypePing() {
 	}
 }
 
+func (h *gqlTWSConnectionHandler) handleMessageTypePong() {
+	h.lastPongReceivedAt = time.Now()
+}
+
+func (h *gqlTWSConnectionHandler) Ping() {
+
+	// Check if we have received a pong message in the pongWaitTimeout
+	// If not we assume the connection is dead and close it
```

Comment:
Will it be confusing to close connections on ping?
I'd assume that a ping func is side effect free.

Reply:
That said, I like your approach to store the pong time and then check it in ping.
One note, you want to make sure that there's enough time between ping and pong, otherwise we close connections where we weren't able to handle pongs yet.

Reply:
That's why I start measuring the time after sending the pong successful or not. Then it depends on how fast we can consume messages from the subgraph if the PONG event arrives within timeout. I think this is safe enough as long as we don't work with a too small timeout.